### PR TITLE
Full review of deferred loading and above-the-folder loading

### PR DESCRIPTION
**Pull request by gladius-mtl**
Closes GH-485

### CODE REVIEW

File: _pages/dev/performance/above-the-fold.md [hunk 1]
```diff
@@ -8,21 +8,21 @@ title: Above-the-Fold Loading (DRAFT)
...
 
-Above-the-fold loading requires two important ingredients:
+Above-the-fold loading requires the following important ingredients:
 
-1. Deferred loading, which is a technique to postpone the creation of components _below_ the fold. For more information, see [Deferred Loading]({{ site.baseurl }}{% link _pages/dev/performance/deferred-loading.md %}).
-2. The notion of the _page fold_. The page fold is not static, and differs from device to device, screen to screen and even the size of the browser.
+1. Deferred loading, which is a technique that postpones the creation of components that are "below the fold". For more information, see [Deferred Loading]({{ site.baseurl }}{% link _pages/dev/performance/deferred-loading.md %}).
+2. The notion of the "page fold". The page fold is not static, and differs from device to device, from screen to screen, and even changes depending on the size of the browser.
 3. A couple of CSS rules to control the page fold.
```
Comment (tobi-or-not-tobi) [inline _pages/dev/performance/above-the-fold.md]: We should rephrase this a bit:
> A couple of CSS rules that moves components initially below the page fold.

File: _pages/dev/performance/above-the-fold.md [hunk 2]
```diff
@@ -8,21 +8,21 @@ title: Above-the-Fold Loading (DRAFT)
...
 
 ## Page Fold Configuration
 
-The page fold is configurable per page template and breakpoint. The page fold configuration is only an indication to speed up the initial creation of page slots above the fold. Page slots are eventually rendered if they're above the fold. The indicated page slot however will be used to prioritize all above the fold page slots over page slots below the fold.
+The page fold is configurable for each page template and breakpoint. The page fold configuration is only an indication to speed up the initial creation of page slots that are above the fold. All page slots are eventually rendered if they are above the fold. You designate the page fold by assigning a page slot to the `pageFold` parameter. This page slot, and all page slots above it, are "above the fold". These page slots are prioritized ahead of page slots that are "below the fold".
```
Comment (tobi-or-not-tobi) [inline _pages/dev/performance/above-the-fold.md]: > All page slots are eventually rendered if they are above the fold

Can we rephrase this to something like:
"All page slots are eventually rendered if they happen to be _above_ the fold"

Comment (tobi-or-not-tobi) [reply _pages/dev/performance/above-the-fold.md]: > This page slot, and all page slots above it

what about: 
"This page slot, and all previous sibling page slots, ..."

File: _pages/dev/performance/above-the-fold.md [hunk 3]
```diff
@@ -8,21 +8,21 @@ title: Above-the-Fold Loading (DRAFT)
...
 
-The page fold is added to the `LayoutConfig` configuration. The page fold indicates the last page slot that should be rendered above the fold.
+The page fold is part of the `LayoutConfig` configuration. The page fold indicates the last page slot that should be rendered above the fold.
```
Comment (tobi-or-not-tobi) [inline _pages/dev/performance/above-the-fold.md]: would be nice to link to https://sap.github.io/cloud-commerce-spartacus-storefront-docs/page-layout/#configuring-the-layout

File: _pages/dev/performance/above-the-fold.md [hunk 4]
```diff
@@ -56,17 +56,17 @@ ProductDetailsPageTemplate: {
 
 ## CSS Configuration
 
-By default, when page slots are loaded on the page, there's no minimum height of page slots or components available. Page slots will initially all have no height, which brings them all in the view port. This eliminates the concept of deferred loading, which is based on content not being the viewport.
+By default, when page slots are loaded on the page, there is no minimum height available for the page slots or components. All page slots initially have no height, which brings them all into the viewport. This prevents the deferred loading technique from working, because it depends on content not being in the viewport.
```
Comment (tobi-or-not-tobi) [inline _pages/dev/performance/above-the-fold.md]: I like a little better explanation here, something like:

> By default, when page slots are loaded on the page, there is no minimum height available for the page slots or components. The actual height is only added when components are loaded, and the associated CSS rules are applied to the components. The page slot will adjust its height automatically when components are loaded. Therefor, page slots do not have an initial height, which is why they end up in the viewport initially. This prevents the deferred loading technique from working, because it depends on content not being in the viewport.

File: _pages/dev/performance/above-the-fold.md [hunk 5]
```diff
@@ -56,17 +56,17 @@ ProductDetailsPageTemplate: {
...
 
-Given that content can be added at runtime, it is not possible to implement a (hard-coded) minimum heights for page slots or components; it all depends on what the business will add at runtime.
+Given that content can be added at runtime, it is not possible to implement a (hard-coded) minimum height for page slots or components – it all depends on what the business will add at runtime.
 
-While this lack of minimum height could be filled up by so-called _ghost design_ CSS rules, there will always be a gap between the ghost design and the actual content.
+While this lack of minimum height could be filled up by so-called "ghost design" CSS rules, there will always be a gap between the ghost design and the actual content.
```
Comment (tobi-or-not-tobi) [inline _pages/dev/performance/above-the-fold.md]: can we add the following to the end. 
Moreover, "ghost design" rules is taking an implementation effort and might not be available.

File: _pages/dev/performance/deferred-loading.md [hunk 6]
```diff
@@ -8,24 +8,24 @@ title: Deferred Loading (DRAFT)
 
 {% include docs/feature_version.html content=version_note %}
 
-Deferred Loading is a technique to postpone rendering of CMS Components. Components which are outside the user's visible area (the *viewport*), will not be rendered in advance. This technique is introduced to improve the initial rendering of the complete page. This does not only saves the creating of components but also defers further loading of component specific data.
+Deferred loading is a technique that is used to postpone the rendering of CMS components. Any component that is outside the viewport (that is, the area that is visible to the user) is not rendered in advance. This improves the initial rendering of the complete page. Deferred loading not only defers the creation of components, it also defers the loading of component-specific data.
```
Comment (tobi-or-not-tobi) [inline _pages/dev/performance/deferred-loading.md]: "it also defers the loading of component-specific data."
can we change that to:

"it also defers the loading of of any component specific data such as media or back end calls ."

File: _pages/dev/performance/deferred-loading.md [hunk 7]
```diff
@@ -56,16 +56,18 @@ ConfigModule.withConfig(<CmsConfig>{
...
--   Above-the-fold loading – Deferred Loading is tightly coupled to so-called [Above-the-Fold Loading]({{ site.baseurl }}{% link _pages/dev/performance/above-the-fold.md %}), but this is documented in another section.
--   Deferred Loading is not related to the native lazy loading of images which has recently landed in Chrome. This new API is not yet fully supported, but could be used in combination with deferred loading for CMS components.
+- Lazy loading of JS chunks: Deferred loading does not lazy-load the component JS bundles. Deferred loading is only concerned with the creation or rendering of components.
+
+- Above-the-fold loading: Deferred loading is tightly coupled to so-called "above-the-fold loading". For more information, see [Above-the-Fold Loading]({{ site.baseurl }}{% link _pages/dev/performance/above-the-fold.md %}).
+
+- Native lazy loading of images: Deferred loading is not related to the native lazy loading of images that has recently been implemented in Chrome. This new API is not yet fully supported, but could be used in combination with deferred loading for CMS components.
```
Comment (tobi-or-not-tobi) [inline _pages/dev/performance/deferred-loading.md]: can we change this to 
" that has recently been implemented in some browsers. "

File: _pages/dev/performance/above-the-fold.md (hunk 1 above)
Comment (gladius-mtl) [reply _pages/dev/performance/above-the-fold.md]: Done.

File: _pages/dev/performance/above-the-fold.md (hunk 2 above)
Comment (gladius-mtl) [reply _pages/dev/performance/above-the-fold.md]: Done.

File: _pages/dev/performance/above-the-fold.md (hunk 3 above)
Comment (gladius-mtl) [reply _pages/dev/performance/above-the-fold.md]: Done.

File: _pages/dev/performance/above-the-fold.md (hunk 4 above)
Comment (gladius-mtl) [reply _pages/dev/performance/above-the-fold.md]: Very nice!
Done.

File: _pages/dev/performance/deferred-loading.md (hunk 6 above)
Comment (gladius-mtl) [reply _pages/dev/performance/deferred-loading.md]: Done.

File: _pages/dev/performance/deferred-loading.md (hunk 7 above)
Comment (gladius-mtl) [reply _pages/dev/performance/deferred-loading.md]: Done.

File: _pages/dev/performance/above-the-fold.md (hunk 5 above)
Comment (gladius-mtl) [reply _pages/dev/performance/above-the-fold.md]: Done.
